src/components/ArticlePreview.jsx

```diff
@@ -0,0 +1,51 @@
+import { Link } from "react-router-dom";
+import { useFavorite } from "../hooks/useFavorite";
+import { formatDate } from "../utils/date";
+
+const ArticlePreview = ({ article }) => {
+  const articleDate = formatDate(article.createdAt);
+  const { handleToggleFavorite, favoritesCount, favoritedButtonBackground } =
+    useFavorite(article);
+
+  return (
+    <div className="article-preview">
+      <div className="article-meta">
+        <Link to={`/@${article.author.username}`}>
+          <img alt="profile" src={article.author.image} />
+        </Link>
+        <div className="info">
+          <Link to={`/@${article.author.username}`} className="author">
+            {article.author.username}
+          </Link>
+          <span className="date">{articleDate}</span>
+        </div>
+        <button
+          className={`btn ${favoritedButtonBackground} btn-sm pull-xs-right`}
+          onClick={handleToggleFavorite}
+        >
+          <i className="ion-heart"></i> {favoritesCount}
+        </button>
+      </div>
+      <Link to={`/article/${article.slug}`} className="preview-link">
+        <h1>{article.title}</h1>
+        <p>{article.description}</p>
+        <div className="display-flex space-between">
+          <span>Read more...</span>
+          {article.tagList.length ? (
+            <div className="tags-container">
+              <ul className="tag-list" id="tag-list">
+                {article.tagList.map((tag) => (
+                  <li key={tag} className="tag-default tag-pill tag-outline">
+                    {tag}
+                  </li>
+                ))}
+              </ul>
+            </div>
+          ) : null}
```

Comment:
aqui vc pode fazer assim pra evitar o ternário:
```suggestion
          {article.tagList.length && (
            <div className="tags-container">
              <ul className="tag-list" id="tag-list">
                {article.tagList.map((tag) => (
                  <li key={tag} className="tag-default tag-pill tag-outline">
                    {tag}
                  </li>
                ))}
              </ul>
            </div>
          ) }
```
Ele já checa se sua condição é verdadeira, e se for, ele renderiza seu componente

Reply:
blz vou soh transformar o `.length` em boolean adicionando `!!` no inicio da variavel, pra resolver esses problemas aq [Stop Using “&&” for Conditional Rendering in React Without Thinking](https://medium.com/geekculture/stop-using-for-conditional-rendering-in-react-a0f7b96200f8). 
Ex:
`!!article.tagList.length && ( ...`